### DISCUSSION TITLE
Mapping data between domains/boxes

### DISCUSSION THREAD

**Discussion by rbielaw** [AMReX-Codes/amrex#3289] [General]
Hello,

I'm trying to figure out the best way to map data between two domains that overlap.  For context, to make in-situ analysis and generation of png files easier, it is desired to have some sub-region of the total domain mapped onto a grid with constant spacing. I can generate a box with the desired domain and spacing but was wondering if there were functions to do the averaging/mapping between the simulation domain and this newly created box? 

Thanks

**Comment by WeiqunZhang** [comment]
Maybe you can take a look at `AmrData::FillVar`. https://github.com/AMReX-Codes/amrex/blob/a3e175871e0ebba2a0fd7a96e85481f1b334285d/Src/Extern/amrdata/AMReX_AmrData.H#L120 There are some examples in amrex/Tools/C_util and amrex/Tools/Postprocessing/C_Src showing how to use DataServices, AmrData and FillVar.

You can also write your own function and use the fillpatch functions FillPatchSingleLevel and FillPatchTwoLevels in AMReX_FillPatchUtil.H.
